pyanaconda/modules/subscription/installation.py

```diff
@@ -0,0 +1,72 @@
+#
+# Copyright (C) 2020 Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+import os
+
+from pyanaconda.core import util
+
+from pyanaconda.modules.common.task import Task
+
+from pyanaconda.modules.common.errors.installation import InsightsConnectError, \
+    InsightsClientMissingError
+
+from pyanaconda.anaconda_loggers import get_module_logger
+log = get_module_logger(__name__)
+
+class ConnectToInsightsTask(Task):
+    """Connect the target system to Red Hat Insights."""
+
+    INSIGHTS_TOOL_PATH = "/usr/bin/insights-client"
+
+    def __init__(self, sysroot, subscription_attached, connect_to_insights):
+        """Create a new task.
+
+        :param str sysroot: target system root path
+        :param bool subscription_attached: if True then the system has been subscribed, False otherwise
+        :param bool connect_to_insights: if True then connect the system to Insights, if False do nothing
+        """
+        super().__init__()
+        self._sysroot = sysroot
+        self._subscription_attached = subscription_attached
+        self._connect_to_insights = connect_to_insights
+
+    @property
+    def name(self):
+        return "Connect the target system to Red Hat Insights"
+
+    def run(self):
+        """Connect the target system to Red Hat Insights."""
+        # check if we should connect to Red Hat Insights
+        if not self._connect_to_insights:
+            log.debug("insights-connect-task: Insights not requested, skipping")
+            return
+        elif not self._subscription_attached:
+            log.debug("insights-connect-task: Insights requested but target system is not subscribed, skipping")
+            return
+
+        # drop the leading / from INSIGHTS_TOOL_PATH to avoid issues with
+        # os.path.join() due to two leading / in paths
+        insights_path = os.path.join(self._sysroot, self.INSIGHTS_TOOL_PATH[1:])
```

Comment:
Please, use `join_paths` from https://github.com/rhinstaller/anaconda/pull/2426.

Reply:
Yep, will do once that PR is merged.

Reply:
It's merged.

Reply:
Thanks! :)